prow/cmd/tackle/main.go

```diff
@@ -0,0 +1,729 @@
+
+// contextConfig returns the loader and config, which can create a clientconfig.
+func contextConfig() (clientcmd.ClientConfigLoader, *clientcmdapi.Config, error) {
+	if err := ensureKubectl(); err != nil {
+		fmt.Println("Prow's tackler requires kubectl, please install:")
+		fmt.Println("  *", err)
+		if gerr := ensureGcloud(); gerr != nil {
+			fmt.Println("  *", gerr)
+		}
+		return nil, nil, errors.New("missing kubectl")
+	}
+
+	l := clientcmd.NewDefaultClientConfigLoadingRules()
+	c, err := l.Load()
+	return l, c, err
+}
+
+// selectContext allows the user to choose a context
+// This may involve creating a cluster
+func selectContext(cfg *clientcmdapi.Config) (string, error) {
+	// list contexts and ask to user to choose a context
+	options := map[int]string{}
+	idx := 0
+	var def string
+
+	fmt.Println("Existing kubectl contexts:")
+	for ctx := range cfg.Contexts {
+		if def == "" || ctx == cfg.CurrentContext {
+			def = ctx
+		}
+		options[idx] = ctx
+		fmt.Printf("  %d: %s", idx, ctx)
+		fmt.Println()
+		idx++
+	}
+	fmt.Println()
+	fmt.Printf("Choose (or create) context [%s]: ", def)
+	var choice string
+	fmt.Scanln(&choice)
+	if choice == "" {
+		choice = def
+	}
+
+	if choice == "create" {
+		ctx, err := createContext()
+		if err != nil {
+			return "", fmt.Errorf("create context: %v", err)
+		}
+		return ctx, nil
+	}
+
+	if _, ok := cfg.Contexts[choice]; ok {
+		return choice, nil
+	}
+
+	var err error
+	if idx, err = strconv.Atoi(choice); err != nil {
+		return "", fmt.Errorf("invalid context: %q", choice)
+	}
+
+	if ctx, ok := options[idx]; ok {
+		return ctx, nil
+	}
+
+	return "", fmt.Errorf("invalid index: %d", idx)
+}
+
+// applyCreate will
+func applyCreate(ctx string, args ...string) error {
+	create := exec.Command("kubectl", append([]string{"--dry-run=true", "--output=yaml", "create"}, args...)...)
+	create.Stderr = os.Stderr
+	obj, err := create.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("rolebinding pipe: %v", err)
+	}
+
+	if err := create.Start(); err != nil {
+		return fmt.Errorf("start create: %v", err)
+	}
+	if err := apply(ctx, obj); err != nil {
+		return fmt.Errorf("apply: %v", err)
+	}
+	if err := create.Wait(); err != nil {
+		return fmt.Errorf("create: %v", err)
+	}
+	return nil
+}
+
+func apply(ctx string, in io.Reader) error {
+	apply := exec.Command("kubectl", "--context="+ctx, "apply", "-f", "-")
+	apply.Stderr = os.Stderr
+	apply.Stdout = os.Stdout
+	apply.Stdin = in
+	if err := apply.Start(); err != nil {
+		return fmt.Errorf("start: %v", err)
+	}
+	return apply.Wait()
+}
+
+func applyRoleBinding(context string) error {
+	fmt.Println("Applying role binding...")
+	who, err := currentAccount()
+	if err != nil {
+		return fmt.Errorf("current account: %v", err)
+	}
+	return applyCreate(context, "clusterrolebinding", "prow-admin", "--clusterrole=cluster-admin", "--user="+who)
+}
+
+func githubToken() (string, error) {
+	fmt.Print("Input /path/to/github/token: ")
+	var path string
+	fmt.Scanln(&path)
```

Comment:
Mind if I add that in a follow-up PR?

Yes, I want to have flags for all these values, and if they aren't present then it will go out to stdin to ask for them.

Reply:
SGTM